File: testcases/benchmark/fibonacci.f90
integer function fibonacci(n)
    integer, intent(in) :: n
    integer :: prev, curr, nxt, i
    if (n <= 1) then
        fibonacci = n
        return
    end if
    prev = 0
    curr = 1
    do i = 2, n
        nxt = prev + curr
        prev = curr
        curr = nxt
    end do
    fibonacci = curr
end function fibonacci

File: testcases/benchmark/fibonacci.rs
#[no_mangle]
pub fn fibonacci(n: i32) -> i64 {
    if n <= 1 { return n as i64; }
    let mut prev = 0;
    let mut curr = 1;
    for _ in 2..=n {
        let next = prev + curr;
        prev = curr;
        curr = next;
    }
    curr
}